### ISSUE TITLE
Standalone wine builds in AppImage format

### ISSUE THREAD

**Issue by probonopd** [PhoenicisOrg/phoenicis-winebuild#128]
(Continuation from https://github.com/ferion11/Proton_Appimage/issues/2#issuecomment-667522302)

Providing standalone wine builds in [AppImage](https://appimage.org/) format would have several advantages.

In a nutshell, we would have a single executable file that could run w32 apps on 64-bit Linux systems without the need for any 32-bit libraries to be installed on the target system.

Proof-of-concept:
* https://github.com/Hackerl/Wine_Appimage/releases/tag/continuous
* https://github.com/ferion11/Proton_Appimage/releases/tag/continuous

In a second step, we could use the same methodology to package standalone w32 apps as single executable files using the AppImage format (maybe automated using https://github.com/PhoenicisOrg/scripts) reusing much of the same work.

Proof-of-concept:
* https://github.com/probonopd/libhookexecv/releases/tag/continuous

@qparis [here](https://github.com/ferion11/Proton_Appimage/issues/2#issuecomment-667511971):

>  Phoenicis wine build could automatically build any version to an app image format. What do we need to know to create a wine app image from a source root?

Looking at https://github.com/Hackerl/Wine_Appimage/blob/master/deployscript/debian-winedeploy.sh, it is actually quite easy - the existing wine builds can be reused, and an AppImage can be made by just executing a few extra steps:

1.  Add `libhookexecv.so` and `wine-preloader_hook` [here](https://github.com/Hackerl/Wine_Appimage/blob/144ea0d42a7e320837b597ed969731a369003863/deployscript/debian-winedeploy.sh#L15-L18)
2. Add all dependencies of wine (in this example, using debian) [here](https://github.com/Hackerl/Wine_Appimage/blob/144ea0d42a7e320837b597ed969731a369003863/deployscript/debian-winedeploy.sh#L20-L27)
3. Turn the directory into an AppImage using appimagetool [here](https://github.com/Hackerl/Wine_Appimage/blob/144ea0d42a7e320837b597ed969731a369003863/deployscript/debian-winedeploy.sh#L29-L42)

**Comment by probonopd** [comment]
In case you wonder what `libhookexecv.so` and `wine-preloader_hook` are all about, it is explained in detail at 

* https://www.winehq.org/pipermail/wine-devel/2017-November/119906.html
* https://www.winehq.org/pipermail/wine-devel/2017-November/119944.html

Instead of getting a patch into wine upstream, @Hackerl made a solution using `LD_PRELOAD`: https://github.com/Hackerl/Wine_Appimage/issues/11.

If you think it would be cleaner, it might still be worthwhile to get the solution into wine upstream proper. Since @qparis is a well-known wine community member, I guess that with your support it could be achieved.

**Comment by qparis** [comment]
Ok ! It seems quite easy. In fact, if we start from the .tar.gz generated by winebuild, the 2) is already done.
  
I suggest we start with wine-preloader_hook to see if it works before implementing some patches. 

Simple question: can the appimagetool run inside a docker container?

**Comment by probonopd** [comment]
Yes. It comes as an AppImage itself which means it needs FUSE. For security reasons, FUSE is usually disabled in Docker containers, so you can use `--appimage-extract` or `--appimage-extract-and-run` as described at https://github.com/AppImage/AppImageKit/wiki/FUSE#docker:

```
wget -c https://github.com/AppImage/AppImageKit/releases/download/continuous/appimagetool-x86_64.AppImage
chmod +x ./appimagetool-x86_64.AppImage
chmod +x ./appimagetool-x86_64.AppImage --appimage-extract-and-run <other command line arguments>
```
